crypto/src/hash/poseidon.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use ark_ff::{biginteger::BigInteger256, FromBytes};
+use mina_curves::pasta::*;
+use oracle::poseidon::*;
+
+/// Converts prefix string into scalar type of vesta elliptic curve
+/// Note that Fp is the base field type of pallas elliptic curve and
+/// the scalar field type of vesta elliptic curve
+pub fn prefix_to_field(s: &[u8]) -> Result<Fp, std::io::Error> {
+    const LEN: usize = 32;
+    let mut bytes = [0_u8; LEN];
+    for (i, &b) in s.iter().enumerate().take(LEN) {
+        bytes[i] = b;
+    }
+
+    let big = BigInteger256::read(bytes.as_slice())?;
```

Comment:
```suggestion
    let big = BigInteger256::read(&s[0..LEN])?;
```

Reply:
This works right? I think `read` is going its own copy internally so there was no need for the extra one in this function

Reply:
Or is padded with zeros the desired result? In that case maybe add a comment explaining why

Reply:
No big256 read panics when input byte slice is less than 256 bits, will add comments here

Reply:
The padding logic is equivalent to [this](https://github.com/o1-labs/snarky/blob/f4c6ba7c3c71bdf5378636d022d11474cd92adb1/fold_lib/fold.ml#L101)

Reply:
Got it. Thanks for clarifying